### PR TITLE
Upgrade mstflint to 4.26.0-1 from 4.21.0-1

### PR DESCRIPTION
**Pull request by jcamposeco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrade mstflint package to v4.26.0-1

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=493529&view=results

### CODE REVIEW

File: SPECS/mstflint/mstflint.spec [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+Release:        1%{?dist}
 License:        GPLv2 OR BSD
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/System
 URL:            https://github.com/Mellanox/%{name}
-Source0:        https://github.com/Mellanox/%{name}/releases/download/v%{version}-1/%{name}-%{version}-1.tar.gz
-Patch4:         add-default-link-flags-for-shared-libraries.patch
-Patch6:         replace-mlxfwreset-with-mstfwreset-in-mstflint-message.patch
+Source0:        https://github.com/Mellanox/%{name}/releases/download/v%{version}-%{release}/%{name}-%{version}-%{release}.tar.gz
```
Comment (reubeno) [inline SPECS/mstflint/mstflint.spec:10]: It looks like `-1` is part of the upstream URI to this source archive. Note that we need the ability to revise this .spec even if that upstream version doesn't change -- at which point the release would become `2%{?dist}` and the URI won't resolve, no?

**Comment by reubeno** [comment]
Looks like a number of changes to the spec; did we originate them or did we get them from somewhere else?

File: SPECS/mstflint/mstflint.spec (hunk 1 above)
Comment (jcamposeco) [reply SPECS/mstflint/mstflint.spec:10]: yeah, make sense, seems I tried to follow their logic, but is not applicable on our side. Thanks for pointing this out.

**Comment by jcamposeco** [comment]
> Looks like a number of changes to the spec; did we originate them or did we get them from somewhere else?

Everything is coming from the updated spec, I also left out a bunch of supported features, which seems we do not need. Let me know if you want to have a quick look to what is left out.

**Comment by reubeno** [comment]
> > Looks like a number of changes to the spec; did we originate them or did we get them from somewhere else?
> 
> Everything is coming from the updated spec, I also left out a bunch of supported features, which seems we do not need. Let me know if you want to have a quick look to what is left out.

Where is the updated spec you're referencing? Does the upstream OSS project have an RPM-style .spec themselves or did it come from another distro? Some of the changes seem to be steps backwards (e.g., %setup => %autsetup, %make_build => make), hence my questions....

File: SPECS/mstflint/mstflint.spec [hunk 2]
```diff
@@ -30,34 +29,105 @@ This package contains firmware update tool, vpd dump and register dump tools
...
+%{_bindir}/mstvpd
+%{_bindir}/mstmcra
+%{_bindir}/mstconfig
+%{_bindir}/mstfwreset
+%{_bindir}/mstcongestion
+%{_bindir}/mstprivhost
+%{_bindir}/mstfwtrace
+%{_bindir}/mstresourcedump
+%{_bindir}/mstresourceparse
+ %{_bindir}/mstfwmanager
```
Comment (anphel31) [inline SPECS/mstflint/mstflint.spec:72]: There are blank spaces on this line, and at the start of a few other lines. Can you please remove them?

File: SPECS/mstflint/mstflint.spec [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 License:        GPLv2 OR BSD
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/System
 URL:            https://github.com/Mellanox/%{name}
-Source0:        https://github.com/Mellanox/%{name}/releases/download/v%{version}-1/%{name}-%{version}-1.tar.gz
-Patch4:         add-default-link-flags-for-shared-libraries.patch
-Patch6:         replace-mlxfwreset-with-mstfwreset-in-mstflint-message.patch
+Source0:        https://github.com/Mellanox/%{name}/releases/download/v%{version}-%{release}/%{name}-%{version}-%{release}.tar.gz
+ExclusiveArch: i386 i486 i586 i686 x86_64 ia64 ppc ppc64 ppc64le arm64 aarch64
```
Comment (anphel31) [inline SPECS/mstflint/mstflint.spec:11]: This could be reduced to the architectures we actually support, `x86_64` and `aarch64`

**Review by vinceaperri — COMMENTED** [review]
Would this upgrade affect the mstflint driver in kernel-hci?

**Comment by jcamposeco** [comment]
> > > Looks like a number of changes to the spec; did we originate them or did we get them from somewhere else?
> > 
> > 
> > Everything is coming from the updated spec, I also left out a bunch of supported features, which seems we do not need. Let me know if you want to have a quick look to what is left out.
> 
> Where is the updated spec you're referencing? Does the upstream OSS project have an RPM-style .spec themselves or did it come from another distro? Some of the changes seem to be steps backwards (e.g., %setup => %autsetup, %make_build => make), hence my questions....

@reubeno This is the source of the spec file https://github.com/Mellanox/mstflint/blob/v4.26.0-1/mstflint.spec.in

**Comment by jcamposeco** [comment]
> Would this upgrade affect the mstflint driver in kernel-hci?

@vinceaperri I do not see any comment that is not backward compatible in the Release notes, everything that is working is expected to keep working :) Link to Release Notes: https://github.com/Mellanox/mstflint/files/13313191/MSTFLINT.Package.-Firmware.Burning.and.Diagnostics.Tools.Documentation.v4.26.pdf

File: SPECS/mstflint/mstflint.spec [hunk 4]
```diff
@@ -29,35 +28,114 @@ Requires:       python3
...
+%{_includedir}/mstflint/mtcr_com_defs.h
+%{_includedir}/mstflint/mtcr_mf.h
+%{_includedir}/mstflint/tools_layouts/adb_to_c_utils.h
+%{_includedir}/mstflint/tools_layouts/icmd_layouts.h
+%{_includedir}/mtcr_ul/mtcr.h
+%{_includedir}/mtcr_ul/mtcr_com_defs.h
+%{_libdir}/mstflint/libmtcr_ul.a
+%{_libdir}/libmtcr_ul.a
+
+%{mstflint_python_tools}/tools_version.py
```
Comment (reubeno) [inline SPECS/mstflint/mstflint.spec:99]: Are these files *all* development content? The `.h` and `.a` ones seem to be, but not sure about the `.py` and `.so` files below.

File: SPECS/mstflint/mstflint.spec [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
+Release:        1%{?dist}
 License:        GPLv2 OR BSD
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          Applications/System
 URL:            https://github.com/Mellanox/%{name}
 Source0:        https://github.com/Mellanox/%{name}/releases/download/v%{version}-1/%{name}-%{version}-1.tar.gz
-Patch4:         add-default-link-flags-for-shared-libraries.patch
-Patch6:         replace-mlxfwreset-with-mstfwreset-in-mstflint-message.patch
+ExclusiveArch: arm64 aarch64
```
Comment (reubeno) [inline SPECS/mstflint/mstflint.spec:11]: Looks like we used to build this on/for x86_64 and aarch64 before; is that an intentional change?

File: SPECS/mstflint/mstflint.spec [hunk 6]
```diff
@@ -29,35 +30,114 @@ Requires:       python3
...
+MSTFLINT_VERSION_STR="%{name} %{version}-%{release}"
+%configure --enable-fw-mgr --enable-adb-generic-tools MSTFLINT_VERSION_STR="${MSTFLINT_VERSION_STR}"
 %make_build
 
 %install
-%make_install
-# Remove the devel files that we don't ship
-rm -fr %{buildroot}%{_includedir}
-find %{buildroot} -type f -name "*.la" -delete -print
-find %{buildroot} -type f -name '*.a' -delete
```
Comment (adithyaj) [inline SPECS/mstflint/mstflint.spec:48]: I don't think we want to add static libraries in general, unless we have a need for them can we remove them?

File: SPECS/mstflint/mstflint.spec [hunk 7]
```diff
@@ -29,35 +30,114 @@ Requires:       python3
...
+%{_includedir}/mstflint/cmdif/icmd_cif_open.h
+%{_includedir}/mstflint/common/compatibility.h
+%{_includedir}/mstflint/mtcr.h
+%{_includedir}/mstflint/mtcr_com_defs.h
+%{_includedir}/mstflint/mtcr_mf.h
+%{_includedir}/mstflint/tools_layouts/adb_to_c_utils.h
+%{_includedir}/mstflint/tools_layouts/icmd_layouts.h
+%{_includedir}/mtcr_ul/mtcr.h
+%{_includedir}/mtcr_ul/mtcr_com_defs.h
+%{_libdir}/mstflint/libmtcr_ul.a
```
Comment (adithyaj) [inline SPECS/mstflint/mstflint.spec:98]: Same as comment above, can we remove static libs?

File: SPECS/mstflint/mstflint.spec (hunk 4 above)
Comment (adithyaj) [reply SPECS/mstflint/mstflint.spec:101]: +1 , do we need the .py files for mstflint?

**Pull request closed without merging**